libc/ctype.c

```diff
@@ -0,0 +1,366 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - ctype.h
+
+ * TESTED:
+ *    - isalnum(), - isalpha(), - isascii(), - isblank(),
+ *    - iscntrl(), - isdigit(), - isgraph(), - islower(),
+ *    - isprint(), - ispunct(), - isspace(), - isupper(),
+ *    - isxdigit(), - toascii(), - tolower(), - toupper(),
+ *    - _toupper(), - _tolower()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <ctype.h>
+#include <stdlib.h>
+#include <time.h>
+#include <unity_fixture.h>
+
+
+#define MIN_VALUE 0
+#define MAX_VALUE 255
+
+/*
+Test group for:
+* isalnum, isalpha, isascii,
+* isblank, iscntrl, isdigit
+* isgraph, islower, isprint
+* ispunct, isspace, isupper
+* isxdigit, toascii, tolower
+* toupper, _tolower, _toupper
+*/
+
+
+TEST_GROUP(ctype);
+
+
+TEST_SETUP(ctype)
+{
+}
+
+
+TEST_TEAR_DOWN(ctype)
+{
+}
+
+
+TEST(ctype, isalnum)
+{
+	for (int i = MIN_VALUE; i <= MAX_VALUE; i++) {
+		if ((i >= '0' && i <= '9') || (i >= 'A' && i <= 'Z') || (i >= 'a' && i <= 'z')) {
+			TEST_ASSERT_TRUE(isalnum(i));
+		}
+		else {
+			TEST_ASSERT_FALSE(isalnum(i));
+		}
+	}
+
+	TEST_ASSERT_FALSE(isalnum(EOF));
+}
+
+
+TEST(ctype, isalpha)
+{
+	for (int i = MIN_VALUE; i <= MAX_VALUE; i++) {
+		if ((i >= 'A' && i <= 'Z') || (i >= 'a' && i <= 'z')) {
+			TEST_ASSERT_TRUE(isalpha(i));
+		}
+		else {
+			TEST_ASSERT_FALSE(isalpha(i));
+		}
+	}
+
+	TEST_ASSERT_FALSE(isalpha(EOF));
+}
+
+
+TEST(ctype, isascii)
+{
+#ifdef __phoenix__  // isascii not yet implemented in Phoenix
+	TEST_IGNORE();
+#endif
+
+#ifndef __phoenix__
```

Comment:
Not needed as there is TEST_IGNORE()

```suggestion
#ifndef __phoenix__
```